tsdb/index/postings.go

```diff
@@ -49,46 +50,63 @@ var ensureOrderBatchPool = sync.Pool{
 	},
 }
 
+const shardCount = 256
+
+type memPostingMap struct {
+	mtx    *sync.RWMutex
```

Comment:
Why is this a pointer?

Reply:
It'a pointer so that we don't copy the mutex when we iterate over all shards in places where we use `for _, shard := range p.m {...}`